packages/tabs/src/TabsOverflow.ts

```diff
@@ -83,6 +90,15 @@ export class TabsOverflow extends SizedMixin(SpectrumElement) {
             tabs.enableTabsScroll = true;
         }
         this.resizeController.observe(this.overflowContainer);
+
+        if (this.autoscroll && tabs?.selected) {
```

Comment:
This is a clear implementation of the described use case. Related question, should there be API/handling so that were the `selected` value of the `<sp-tabs>` element updated from the outside to include content what was not currently visible that this happened as well? Even if it makes sense, I can see separating the work for the two cases, but want to make sure we think through the whole workflow in case there would be any differences in the API application in that case.

Reply:
Now that you mentioned it I think would be more extensible to have a public method on `Tabs`: 
```
    public scrollToSelection(): void {
        if (this.enableTabsScroll && this.selected) {
            this.updateComplete.then(() => {
                const selectedTab = this.querySelector(
                    `[role="tab"][value="${this.selected}"]`
                ) as Tab;
                selectedTab?.scrollIntoView({ inline: 'center', behavior: 'smooth' });
            });
        }
    }
```
similar to the `scrollTo` one which consumers can call if needed.  It can even receive scrollOptions parameters.

The `autoscroll` (or `autoscroll-to-selection`?) property on `TabsOverflow` would just enable this behaviour for you  (would pass a boolean similar to `enableTabsScroll`) and call `scrollToSelection` on `updated`.

```
    protected override updated(changedProperties: PropertyValueMap<this>): void {
        super.updated(changedProperties);
        if (this.shouldAutoScroll && changedProperties.has('selected')) {
           // do check if it is out of view first and then:
            this.scrollToSelection();
        }
    }
```


For this automatic behaviour you would need to calculate the boundings though, so that you don't do unnecessary scrolls on tab click. But I think this enables out of the box scrolling, and adds an API which can be used for custom use cases.

Attaching a video for demonstration, I don't have a strong opinion on the API, so open to any suggestions!


https://github.com/adobe/spectrum-web-components/assets/13311865/f3e9df29-f88e-4fa5-a376-8de36dc20966

Reply:
Does the use of `scrollIntoView` require the use of `_isSelectionOutOfViewport`? I like the direction this is going in, but I'm a but remiss to add DOM measurement to the workflow here...

Reply:
@Westbrook I can totally get rid of the DOM measurement checks, but then the Tabs will always try to scroll even if not necessary

Reply:
Hrm. What if we use the following CSS on the `sp-tabs`:
```
:host {
    scroll-margin-inline: var(--mod-tabs-item-horizontal-spacing, var(--spectrum-tabs-item-horizontal-spacing) );
}
```
And `selectedTab?.scrollIntoView();` with no arguments? Do we get something similar to what we want when a Tab is offscreen and no (I think no, but I'd accept a very small amount, too) scrolling when the element is in view?